documentation/src/docs/segment-containers.md

```diff
@@ -12,25 +12,25 @@ You may obtain a copy of the License at
-On detecting any changes to the cluster membership we verify and re-map all containers to the available set of pravega hosts. This information is persisted in the HostStore atomically. This is stored as a single blob and contains a Map of Host to Set of Containers that a host owns.
+Each Pravega Controller runs a service which monitors the ephemeral nodes on the Zookeeper and detects all active Pravega hosts in the cluster.
+If any changes are detected to the cluster membership, then all Containers are verified and re-mapped to the available set of Pravega hosts. This information is persisted in the HostStore atomically. This is stored as a single blob and contains a map of Host to set of Containers that a host owns.
 
-We use zookeeper to ensure only one pravega controller is monitoring the cluster to avoid multiple simultaneous writers to the HostStore. This will avoid race conditions and allow the state to converge faster.
+We use Zookeeper to ensure only one Pravega Controller is monitoring the cluster to avoid multiple simultaneous Writers to the HostStore. This will avoid race conditions and allow the state to converge faster.
 
 ## Rebalance Frequency
-Rebalance of the container ownership happens when either a pravega host is added or removed from the cluster. Since this is a costly operation we need to guard against doing this very often. Currently we ensure that a configured minimum time interval is maintained between any 2 rebalance operations. In the worst case this will proportionally increase the time it takes to perform ownership change in the cluster.
+Rebalance of the Container ownership happens when either a Pravega Host is added or removed from the cluster. Since, this is an expensive operation and not to be repeated very often. Currently we ensure that a configured minimum time interval is maintained between any two rebalance operations. In the worst case, this will proportionally increase the time it takes to perform ownership change in the cluster.
 
 # Ownership Change Notification
-Every pravega host has a long running Segment Manager Service. This constantly polls/watches the HostStore for any changes to the container ownership. On detecting any ownership changes for itself (identified by the host key in the Map) the Segment Manager triggers addition and removal of containers accordingly.
+Every Pravega host has a long running Segment Manager Service. This constantly polls/watches the HostStore for any changes to the container ownership. On detecting any ownership changes for itself (identified by the host key in the Map) the Segment Manager triggers addition and removal of containers accordingly.
 
 ## Ensuring a container is stopped on one host before it is started on another host
 We currently plan to use conservative timeouts on the Pravega host to ensure a container is not started before it is stopped on another node. This needs further review/discussion.
```

Comment:
@andreipaduroiu @fpj 
Any updates on this??

Reply:
no

Reply:
@andrei Cani remove the line "This needs further review/discussion".??

Reply:
yes, remove that. Actually we should probably remove this whole paragraph